File: back_end/back-end/src/main/java/com/flavio/backend/model/object/Pedido.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.flavio.backend.model.object;

import java.io.Serializable;
import java.util.Date;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

/**
 *
 * @author Flavio
 */
@Entity
@Table(name = "pedido")
public class Pedido implements Serializable {
    
    private static final long serialVersionUID = 1L;
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @ManyToOne
    private Funcionario funcionario;
    private double total;
    @ManyToOne
    private Mesa mesa;
    @Temporal(TemporalType.TIMESTAMP)
    private Date dataPedido;
    @Column(length = 20)
    private String status;
    @ManyToMany
    @JoinTable(name = "pedido_cardapio",joinColumns = @JoinColumn(name="pedido_id"),inverseJoinColumns = @JoinColumn(name = "item_cardapio_id"))
    private List<ItemCardapio>itens = new ArrayList<>();

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Funcionario getFuncionario() {
        return funcionario;
    }

    public void setFuncionario(Funcionario funcionario) {
        this.funcionario = funcionario;
    }

    public double getTotal() {
        return total;
    }

    public void setTotal(double total) {
        this.total = total;
    }

    public Mesa getMesa() {
        return mesa;
    }

    public void setMesa(Mesa mesa) {
        this.mesa = mesa;
    }

    public Date getDataPedido() {
        return dataPedido;
    }

    public void setDataPedido(Date dataPedido) {
        this.dataPedido = dataPedido;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public List<ItemCardapio> getItens() {
        return itens;
    }

    public void setItens(List<ItemCardapio> itens) {
        this.itens = itens;
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 67 * hash + this.id;
        hash = 67 * hash + Objects.hashCode(this.funcionario);
        hash = 67 * hash + (int) (Double.doubleToLongBits(this.total) ^ (Double.doubleToLongBits(this.total) >>> 32));
        hash = 67 * hash + Objects.hashCode(this.mesa);
        hash = 67 * hash + Objects.hashCode(this.dataPedido);
        hash = 67 * hash + Objects.hashCode(this.status);
        hash = 67 * hash + Objects.hashCode(this.itens);
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Pedido other = (Pedido) obj;
        if (this.id != other.id) {
            return false;
        }
        if (Double.doubleToLongBits(this.total) != Double.doubleToLongBits(other.total)) {
            return false;
        }
        if (!Objects.equals(this.status, other.status)) {
            return false;
        }
        if (!Objects.equals(this.funcionario, other.funcionario)) {
            return false;
        }
        if (!Objects.equals(this.mesa, other.mesa)) {
            return false;
        }
        if (!Objects.equals(this.dataPedido, other.dataPedido)) {
            return false;
        }
        if (!Objects.equals(this.itens, other.itens)) {
            return false;
        }
        return true;
    }
    
    
    
    
    
}

File: mobile/projeto_gerenciamento_pedido/lib/model/Pedido.dart
import 'package:projeto_gerenciamento_pedido/model/Cardapio.dart';

import 'Funcionario.dart';
import 'Mesa.dart';

class PedidoModel {
  int id;
  Funcionario funcionario;
  double total;
  Mesa mesa;
  String dataPedido;
  String status;
  List<ItemCardapio> itens;

  PedidoModel(
      {this.id,
      this.funcionario,
      this.total,
      this.mesa,
      this.dataPedido,
      this.status,
      this.itens});

  PedidoModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    funcionario = json['funcionario'] != null
        ? new Funcionario.fromJson(json['funcionario'])
        : null;
    total = json['total'];
    mesa = json['mesa'] != null ? new Mesa.fromJson(json['mesa']) : null;
    dataPedido = json['dataPedido'];
    status = json['status'];
    if (json['itens'] != null) {
      itens = new List<ItemCardapio>();
      json['itens'].forEach((v) {
        itens.add(new ItemCardapio.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    if (this.funcionario != null) {
      data['funcionario'] = this.funcionario.toJson();
    }
    data['total'] = this.total;
    if (this.mesa != null) {
      data['mesa'] = this.mesa.toJson();
    }
    data['dataPedido'] = this.dataPedido;
    data['status'] = this.status;
    if (this.itens != null) {
      data['itens'] = this.itens.map((v) => v.toJson()).toList();
    }
    return data;
  }
}
